drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2163 @@
+		/* nRetryCount = 2 for PD REV 3.0 */
+		txctl_retrycnt = 2 << UTCPD_TXCTL_RETRYCNT_Pos;
+	} else if (msg->type <= PD_PACKET_TX_BIST_MODE_2) {
+		/* Per TCPCI spec, no retry for non-SOP* transmission */
+		txctl_retrycnt = 0;
+	} else {
+		LOG_ERR("Invalid PD packet type: %d", msg->type);
+		rc = -EINVAL;
+		goto cleanup;
+	}
+	/* NOTE: Needn't extra cast for UTCPD_TXCTL.TXSTYPE aligning with pd_packet_type */
+	txctl_txstype = ((uint32_t)msg->type) << UTCPD_TXCTL_TXSTYPE_Pos;
+	txctl = txctl_retrycnt | txctl_txstype;
+	NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, TXCTL, txctl);
+
+cleanup:
+
+	return rc;
+}
+
+static int numaker_tcpc_dump_std_reg(const struct device *dev)
+{
+	return numaker_utcpd_dump_regs(dev);
+}
+
+static int numaker_tcpc_get_snk_ctrl(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t pwrsts = NUMAKER_UTCPD_REG_READ_BY_NAME(dev, PWRSTS);
+
+	return (pwrsts & UTCPD_PWRSTS_SKVB_Msk) ? true : false;
+}
+
+static int numaker_tcpc_get_src_ctrl(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	uint32_t pwrsts = NUMAKER_UTCPD_REG_READ_BY_NAME(dev, PWRSTS);
+
+	return (pwrsts & (UTCPD_PWRSTS_SRVB_Msk | UTCPD_PWRSTS_SRHV_Msk)) ? true : false;
+}
+
+static int numaker_tcpc_sop_prime_enable(const struct device *dev, bool enable)
+{
+	struct numaker_tcpc_data *data = dev->data;
+
+	data->rx_sop_prime_enabled = enable;
+
+	return 0;
+}
+
+static int numaker_tcpc_set_bist_test_mode(const struct device *dev, bool enable)
+{
+	return numaker_utcpd_bist_test_mode_set_enable(dev, enable);
+}
+
+static int numaker_tcpc_set_alert_handler_cb(const struct device *dev,
+					     tcpc_alert_handler_cb_t alert_handler,
+					     void *alert_data)
+{
+	struct numaker_tcpc_data *data = dev->data;
+
+	data->tcpc_alert.handler = alert_handler;
+	data->tcpc_alert.data = alert_data;
+
+	return 0;
+}
+
+/* Functions below with name pattern "*_tcpc_ppc_*" are to invoke by NuMaker PPC driver */
+
+int numaker_tcpc_ppc_is_dead_battery_mode(const struct device *dev)
+{
+	return numaker_utcpd_deadbattery_query_enable(dev);
+}
+
+int numaker_tcpc_ppc_exit_dead_battery_mode(const struct device *dev)
+{
+	return numaker_utcpd_deadbattery_set_enable(dev, false);
+}
+
+int numaker_tcpc_ppc_is_vbus_source(const struct device *dev)
+{
+	return numaker_utcpd_vbus_is_source(dev);
+}
+
+int numaker_tcpc_ppc_is_vbus_sink(const struct device *dev)
+{
+	return numaker_utcpd_vbus_is_sink(dev);
+}
+
+int numaker_tcpc_ppc_set_snk_ctrl(const struct device *dev, bool enable)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+	int rc = 0;
+	uint32_t cmd;
+
+	if (enable) {
+		cmd = UTCPD_CMD_SINK_VBUS;
+	} else {
+		cmd = UTCPD_CMD_DISABLE_SINK_VBUS;
+	}
+	NUMAKER_UTCPD_REG_WRITE_BY_NAME(dev, CMD, cmd);
+
+cleanup:
```

Comment:
This ```cleanup:``` labels should be removed along with the other in the file.

Reply:
Originally, `goto cleanup` is embedded in `NUMAKER_UTCPD_REG_WRITE_BY_NAME`. Now it is moved outside.